프론트엔드에 (초기) DB 테이블 인터페이스 추가

somnisomni opened the pull request:
TypeScript로 DB와 데이터 교환 로직 작성 시, 타입 유추를 위한 인터페이스를 추가합니다.

현재까지 논의된 내용을 바탕으로 구현하였으며, (당연하지만) 추후 지속적으로 업데이트 될 수 있습니다.

다음과 같은 형식으로 인터페이스를 임포트하면 됩니다.
```ts
import { IUserDisplay, IUser, IComment, IPost } from "@/interfaces/IDatabaseData";
```

woosukji reviewed (commented):
1. IUser 의 전체 정보가 좀 커서, 아이디랑 이름, 보직, 프로필사진만 포함하는 IUserSimple 같은걸 만들고 그걸 extend해서 현재 IUser 을 만들었으면 합니다. 그리고 IComment.author와 IPost.(author, comments), 그리고 IUser.(followings, followers) 에서 IUserSimple을 사용하게요.
- 서버측에서 적당히 딸린 정보들 모아서 보내주면, 받아서 IUser(Simple) 으로 매핑하는 작업은 프론트에서 하기로 결정
2. 포스트를 like 한 사람들 목록은 포함해야 할 것 같습니다. `likes: Array<IUserSimple>` 이렇게 사용하면 어떨까요?
- OK

somnisomni committed:
Add interface to match DB data

TypeScript로 DB와 데이터 교환 로직 작성 시, 타입 유추를 위한
인터페이스를 추가합니다.

현재까지 논의된 내용을 바탕으로 구현하였으며, (당연하지만) 추후
지속적으로 업데이트 될 수 있습니다.

다음과 같은 형식으로 인터페이스를 임포트하면 됩니다.
```ts
import { IUserDisplay, IUser, IComment, IPost } from "@/interfaces/IDatabaseData";
```

somnisomni commented:
이야기 나눈 내용 반영했습니다. 리뷰 부탁드립니다.

somnisomni committed:
Separate `IPost` interface

`IUser`-`IUserDisplay`의 관계와 비슷하게, 뷰(피드)에서의 표시 목적을 겸한
`IPostDisplay`로 인터페이스를 분리합니다.

woosukji reviewed (approved):
수고많으십니당
